### PR TITLE
release/11-lorawan: add tests to target applications

### PR DESCRIPTION
**Pull request by fjmolinas**
### Contribution description

This PR adds two tests that replicate Task01-04 of item 11-lorawan of the Release test spec. The target applications of the tests are `examples/lorawan` and `tests/pkg_semtech_loramac`.

Default configurations have been added to the application makefile to be able to test over iotlab.

Still missing from spec check:
- [ ] verify duty cycling is respected (I have found that this isnt very deterministic, maybe @aabadie has a suggestion on how to do it)

### Testing procedure

Follow the README for [examples/lorawan/README.md](examples/lorawan/README.md) and [tests/pkg_semtech-loramac/README.md](tests/pkg_semtech-loramac/README.md) under Automatic Test and execute the test for both applications

### Issues/PRs references

Depends on ~~#11917~~

### CODE REVIEW

**Comment by miri64** [comment]
@jia200x @leandrolanzieri can you have a look, please?

File: tests/pkg_semtech-loramac/tests/01-run.py [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+def _check_eeprom(child):
+    # Check if eeprom is supported
+    child.sendline("loramac help")
+    child.expect(r'Usage: loramac \<([\w+\|?]+)\>')
+    return (len(child.match.group(1).split('|')) == 7)
+
+
+def _reboot(child, join):
+    if join == "abp":
+        child.sendline("loramac get ul_ctr")
```
Comment (leandrolanzieri) [inline tests/pkg_semtech-loramac/tests/01-run.py:67]:
```suggestion
        child.sendline("loramac get ul_cnt")
```

File: tests/pkg_semtech-loramac/tests/01-run.py [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+        child.expect(r'Uplink Counter: (\d+)')
+        uplink_counter = int(child.match.group(1))
+    child.sendline("reboot")
+    child.expect_exact("All up, running the shell now")
+    child.expect_exact(">")
+    if join == "abp":
+        _send_line_echo(child, "loramac set ul_ctr {}".format(uplink_counter))
```
Comment (leandrolanzieri) [inline tests/pkg_semtech-loramac/tests/01-run.py:74]:
```suggestion
        _send_line_echo(child, "loramac set ul_cnt {}".format(uplink_counter))
```

**Comment by fjmolinas** [comment]
Rebased and addressed comments.

**Comment by fjmolinas** [comment]
I'll add a whitelist to the tests as well

**Comment by fjmolinas** [comment]
> I'll add a whitelist to the tests as well

Actually non on CI, so doesn't matter.

**Comment by fjmolinas** [comment]
@leandrolanzieri addressed your comments!

**Comment by fjmolinas** [comment]
@leandrolanzieri may I squash?

**Comment by leandrolanzieri** [comment]
> @leandrolanzieri may I squash?

yes!

**Comment by leandrolanzieri** [comment]
Hmm there are some white space errors

**Comment by fjmolinas** [comment]
Thanks for the review @leandrolanzieri!

**Comment by aabadie** [comment]
Nice one, thanks !

**Comment by aabadie** [comment]
This was merged without checking "CI: run tests" before => this test script cannot run on Murdock without a complex setup. I would suggest to disable the run tests on CI.

**Pull request merged**